nifi-commons/nifi-write-ahead-log/src/main/java/org/apache/nifi/wali/HashMapSnapshot.java

```diff
@@ -264,10 +267,11 @@ public synchronized void writeSnapshot(final SnapshotCapture<T> snapshot) throws
         }
 
         // Write to the partial file.
-        try (final FileOutputStream fileOut = new FileOutputStream(getPartialFile());
```

Comment:
My intent was to avoid having to check for a null cipher key at every call site, but as you point out, some of those sites are within a try-with-resources, making the static method that I implemented... questionable.

I've replaced the static method with a ctor and some support methods, and updated all of the call sites with a null check.

Reply:
(in this specific case, however, moving the stream construction back into the try-with-resources declaration still causes test failures)